Source/WebKit/UIProcess/API/haiku/WebView.cpp

```diff
@@ -106,96 +90,55 @@ void BWebView::loadURIRequest(const char* uri)
 
 void BWebView::paintContent()
 {
-    getRenderView()->LockLooper();
-    getRenderView()->Invalidate();
-    getRenderView()->UnlockLooper();
+    fWebViewBase->LockLooper();
+    fWebViewBase->Invalidate();
+    fWebViewBase->UnlockLooper();
 }
 
 WebViewBase* BWebView::getRenderView()
 {
-    return toImpl(fViewPort.get());
+    return fWebViewBase.get();
 }
 
 const char* BWebView::getCurrentURL()
 {
-    return getRenderView()->currentURL();
+    return fWebViewBase->currentURL();
 }
 
 void BWebView::loadURI(BMessage* message)
 {
     const char* uri;
     message->FindString("url", &uri);
-    auto page = WKViewGetPage(fViewPort.get());
-    WKRetainPtr<WKURLRef> wuri;
-    wuri = adoptWK(WKURLCreateWithUTF8CString(uri));
-    WKPageLoadURL(page, wuri.get());
+    fWebViewBase->page()->loadRequest(URL { uri });
 }
 
 void BWebView::goForward()
 {
-    auto page = WKViewGetPage(fViewPort.get());
-    WKPageGoForward(page);
+    fWebViewBase->page()->goForward();
     BMessage message(URL_CHANGE);
     message.AddString("url", BString(getCurrentURL()));
     fAppLooper->PostMessage(&message);
 }
 
 void BWebView::goBackward()
 {
-    auto page = WKViewGetPage(fViewPort.get());
-    WKPageGoBack(page);
+    fWebViewBase->page()->goBack();
     BMessage message(URL_CHANGE);
     message.AddString("url", BString(getCurrentURL()));
     fAppLooper->PostMessage(&message);
 }
 
 void BWebView::stop()
 {
-    auto page = WKViewGetPage(fViewPort.get());
-    WKPageClose(page);
-}
-
-void BWebView::didCommitNavigation(WKPageRef page, WKNavigationRef navigation, WKTypeRef userData, const void* clientInfo)
-{
-    BLooper* looper = ((BWebView*)clientInfo)->getAppLooper();
-    BMessage message(DID_COMMIT_NAVIGATION);
-    looper->PostMessage(&message);
-}
-
-void BWebView::didReceiveServerRedirectForProvisionalNavigation(WKPageRef page, WKNavigationRef navigation, WKTypeRef userData, const void* clientInfo)
-{
-    BLooper* looper = ((BWebView*)clientInfo)->getAppLooper();
-    BMessage message(URL_CHANGE);
-    message.AddString("url", BString(((BWebView*)clientInfo)->getCurrentURL()));
-    looper->PostMessage(&message);
-}
-
-void BWebView::didFinishDocumentLoad(WKPageRef page, WKNavigationRef navigation, WKTypeRef userData, const void* clientInfo)
-{
-}
-
-void BWebView::didFinishNavigation(WKPageRef page, WKNavigationRef navigation, WKTypeRef userData,const void* clientInfo)
-{
-    BLooper* looper = ((BWebView*)clientInfo)->getAppLooper();
-    BMessage message(DID_FINISH_NAVIGATION);
-    looper->PostMessage(&message);
-}
-
-void BWebView::didFailNavigation(WKPageRef page, WKNavigationRef navigation,WKErrorRef, WKTypeRef userData,const void* clientInfo)
-{
-}
-
-void BWebView::didFinishProgress(WKPageRef page,const void* clientInfo)
-{
+    fWebViewBase->page()->close();
 }
 
-double BWebView::didChangeProgress()
+double BWebView::progress()
```

Comment:
the old code was a bit confused, but I think the idea of didChangeProgress is that it gets called automatically when the progress on rendering the page changes, and the browser can then update its progress bar or whatever it uses to show loading progress.

If we have just a method to get the progress, we would have to call it periodically from the browser to see if there is any progress, that isn't great. There is already a DID_CHANGE_PROGRESS message in MiniBrowser, how is that triggered? Can the progress info be included directly into the message instead?